configure.ac

```diff
@@ -5998,6 +6001,48 @@ then
     AM_CFLAGS="$AM_CFLAGS -DHAVE_DH_DEFAULT_PARAMS"
 fi
 
+
+AC_ARG_WITH([max-rsa-bits],
+    [AS_HELP_STRING([--with-max-rsa-bits=number],[number of bits to support for RSA, DH, and DSA keys])],
+    [WITH_MAX_ASSYM_KEY_BITS=$withval],
+    [WITH_MAX_ASSYM_KEY_BITS="$DEFAULT_MAX_ASSYM_KEY_BITS"])
+
+if test -n "$WITH_MAX_ASSYM_KEY_BITS"; then
+   if test "$WITH_MAX_ASSYM_KEY_BITS" -lt 1024 -o "$WITH_MAX_ASSYM_KEY_BITS" -gt 16384; then
+       AC_MSG_ERROR([--with-max-rsa-bits argument must be between 1024 and 16384 inclusive])
+   fi
+   AM_CFLAGS="$AM_CFLAGS -DRSA_MAX_SIZE=$WITH_MAX_ASSYM_KEY_BITS -DFP_MAX_BITS=$((WITH_MAX_ASSYM_KEY_BITS * 2)) -DSP_INT_BITS=$WITH_MAX_ASSYM_KEY_BITS"
+fi
+
+AC_ARG_WITH([max-ecc-bits],
+    [AS_HELP_STRING([--with-max-ecc-bits=number],[number of bits to support for ECC algorithms])],
+    [WITH_MAX_ECC_BITS=$withval],
+    [WITH_MAX_ECC_BITS="$DEFAULT_MAX_ECC_BITS"])
+
+if test -n "$WITH_MAX_ECC_BITS"; then
+   if test "$WITH_MAX_ECC_BITS" -lt 112 -o "$WITH_MAX_ECC_BITS" -gt 2048; then
```

Comment:
How did you come up with 2048 for max ECC? Technically we currently do not support higher than 521 for ECC and 1024 for SAKKE.

Reply:
2048 because nothing broke when I set that value, and it's larger than the 1024 required by SAKKE, but I wanted to set it to something reasonable and largely future-proof, to avoid a user complaining that the build or check fails with `--with-max-ecc-bits=1048572`...

Reply:
1024 would be the max for now. When we have a curve that big then change.

Reply:
whoops, I missed this comment on the previous round of touch-ups.  now done.